pkg/appmanager/resourceConfig.go

```diff
@@ -1742,16 +1740,41 @@ func (appMgr *Manager) createRSConfigFromRoute(
 
 // Copies from an existing config into our new config
 func (rc *ResourceConfig) copyConfig(cfg *ResourceConfig) {
-	var cfgBytes bytes.Buffer
-	err := gob.NewEncoder(&cfgBytes).Encode(cfg)
-	if err != nil {
-		log.Errorf("Couldn't encode ResourceConfig: %v", err)
-		return
-	}
-	err = gob.NewDecoder(&cfgBytes).Decode(&rc)
-	if err != nil {
-		log.Errorf("Could not decode ResourceConfig: %v", err)
-		return
+	// MetaData
+	rc.MetaData = cfg.MetaData
+	// Virtual
+	rc.Virtual = cfg.Virtual
+	// Profiles
+	rc.Virtual.Profiles = make([]ProfileRef, len(cfg.Virtual.Profiles))
+	copy(rc.Virtual.Profiles, cfg.Virtual.Profiles)
+	// Pools
+	rc.Pools = make(Pools, len(cfg.Pools))
+	copy(rc.Pools, cfg.Pools)
+	// Pool Members
+	for i, _ := range rc.Pools {
+		rc.Pools[i].Members = make([]Member, len(cfg.Pools[i].Members))
+		copy(rc.Pools[i].Members, cfg.Pools[i].Members)
```

Comment:
Same thing for rc.Pools[i].MonitorNames (do we need to do a make/copy for it since it's an array?)